src/test/java/com/mongodb/hibernate/jdbc/MongoResultSetTests.java

```diff
@@ -0,0 +1,282 @@
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonInt64;
+import org.bson.BsonNull;
+import org.bson.BsonString;
+import org.bson.types.Decimal128;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MongoResultSetTests {
+    private static final List<String> FIELDS = List.of("id", "title", "author");
+
+    @Mock
+    private MongoCursor<BsonDocument> mongoCursor;
+
+    private MongoResultSet mongoResultSet;
+
+    @BeforeEach
+    void setUp() {
+        mongoResultSet = new MongoResultSet(mongoCursor, FIELDS);
+    }
+
+    @Nested
+    class ClosedTests {
+
+        @Test
+        void testIsIdempotent() throws SQLException {
+            // given
+            mongoResultSet.close();
+
+            // when && then
+            assertDoesNotThrow(() -> mongoResultSet.close());
+        }
+
+        @ParameterizedTest(name = "SQLException is thrown when \"{0}\" is called on a closed MongoResultSet")
+        @MethodSource("getMongoResultSetMethodInvocationsImpactedByClosing")
+        void testCheckClosed(String label, ResultSetMethodInvocation methodInvocation) throws SQLException {
+            // given
+            mongoResultSet.close();
+
+            // when && then
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertEquals("MongoResultSet has been closed", exception.getMessage());
+        }
+
+        private static Stream<Arguments> getMongoResultSetMethodInvocationsImpactedByClosing() {
+            return doGetMongoResultSetMethodInvocationsImpactedByClosing();
+        }
+    }
+
+    @Nested
+    class ColumnIndexCheckingTests {
+
+        @ParameterizedTest(
+                name = "SQLException is thrown when \"{0}\" is called on a MongoResultSet with columnIndex too low")
+        @MethodSource("getMongoResultSetMethodInvocationsWithColumnIndexUnderflow")
+        void testColumnIndexUnderflow(String label, ResultSetMethodInvocation methodInvocation) {
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertTrue(exception.getMessage().startsWith("Invalid column index"));
+        }
+
+        @ParameterizedTest(
+                name = "SQLException is thrown when \"{0}\" is called on a MongoResultSet with columnIndex too high")
+        @MethodSource("getMongoResultSetMethodInvocationsWithColumnIndexOverflow")
+        void testColumnIndexOverflow(String label, ResultSetMethodInvocation methodInvocation) {
```

Comment:
Below, instead of using a methodSource, there is:

```
 assertAll(
                // f1: new BsonNull()
                () -> assertAll(
                        () -> assertNull(mongoResultSet.getString(1)),
                        () -> assertFalse(mongoResultSet.getBoolean(1)),
```

It seems the same could be done here - get the list of methods, then iterate over them in the one test, rather than creating multiple tests per method, since that is what we do below.

Ultimately, I would like to avoid using different (and inconsistently used?) constructs to model test grouping/separation.

Reply:
I hesitated to do this for as long as we used `@Nested`, it is perfectly fine to have different testing patterns in the same class. We had done this in https://github.com/mongodb/mongo-hibernate/blob/main/src/test/java/com/mongodb/hibernate/jdbc/MongoPreparedStatementTests.java.

Not saying it is not changable, but without good reason, it seems doubtable to break consistency across the two JDBC testing classes.